lib/SetAPI/generic/GenericSetNavigator.py

```diff
@@ -110,16 +114,16 @@ def _list_all_sets(self, workspaces, include_metadata):
             print(("Time of ws_info listing: " + str(time.time() - t1)))
 
         t2 = time.time()
-        sets = []
-        processed_refs = {}
-        for t in GenericSetNavigator.SET_TYPES:
-            list_params = {"includeMetadata": include_metadata, "type": t}
+        sets = [
+            {"ref": info_to_ref(s), "info": s}
+            for t in GenericSetNavigator.SET_TYPES
             for s in WorkspaceListObjectsIterator(
-                self.ws, list_objects_params=list_params, ws_info_list=ws_info_list
-            ):
-                ref = self._build_obj_ref(s)
-                sets.append({"ref": ref, "info": s})
-                processed_refs[ref] = True
```

Comment:
nothing is ever done with `processed_refs` so not much point in having it around!